### PR TITLE
Moving from .framework to .xcframework

### PR DESCRIPTION
**Pull request by russelarms**
The main goal is the ability to run projects using this library on Simulator on M1 devices

Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description
Now build-fat-framework.sh creates .xcframework with separated .frameworks for iphoneos and iphonesimulator.

## Related PRs or issues
https://github.com/microsoft/appcenter-sdk-react-native/issues/895
https://github.com/microsoft/appcenter-sdk-react-native/pull/896

## Misc
Podspecs version is bumped from 4.0.2 to 4.0.3 to make local pod installation work.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Version 4.0.3 (Under development)
 
+### App Center
+
+#### iOS
+
+* **[Feature]** Use XCFramework format for the AppCenterReactNativeShared pod.
```
Comment (MatkovIvan) [inline CHANGELOG.md:9]: Let's mention that it fixes Apple Silicon simulators (and probably some release builds)

**Review by MatkovIvan — APPROVED** [review]

**Comment by JamesBond95470** [comment]
Hi thanks for the merge, will there be a release with this changes ?

**Comment by MatkovIvan** [comment]
This quarter, but there is no specific date for the next release.

**Comment by JamesBond95470** [comment]
So we can't expect to have a solution for M1 macbook's for now ?

**Comment by MatkovIvan** [comment]
You can use Rosetta for now. During our tests it's required anyway because CocoaPods itself do not support arm yet

**Pull request merged**